### PR TITLE
Issue #SB-13575 fix: downloading status is showing

### PR DESCRIPTION
**Pull request by swetha-aggidevara**

### CODE REVIEW

File: server/controllers/content/content.ts [hunk 1]
```diff
@@ -416,15 +429,15 @@ export default class Content {
...
+            await this.searchDownloadingContent(listOfContentIds, reqId)
                 .then(data => {
-                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in ContentDb`)
+                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in Content_Download Db`)
                     for (let doc of data.docs) {
                         for (let content of contents) {
-                            logger.debug(`include addedToLibrary property for the contents which are downloaded`)
-                            this.includeAddedToLibraryProperty(doc, content, reqId);
-                            logger.info(`ReqId = "${reqId}": included addedToLibrary property for the contents which are downloaded`)
+                            logger.debug(`add downloadStatus for the contents`)
```
Comment (anoophm) [inline server/controllers/content/content.ts]: remove this debug line

File: server/controllers/content/content.ts [hunk 2]
```diff
@@ -416,15 +429,15 @@ export default class Content {
...
+                            this.includeDownloadStatus(doc, content, reqId);
+                            logger.info(`ReqId = "${reqId}": added downloadStatus for the contents`)
```
Comment (anoophm) [inline server/controllers/content/content.ts]: remove this also

File: server/controllers/content/content.ts [hunk 3]
```diff
@@ -14,9 +14,19 @@ import Hashids from "hashids";
 import { containerAPI } from "OpenRAP/dist/api";
 import * as TreeModel from "tree-model";
 
+export enum DOWNLOAD_STATUS {
```
Comment (anoophm) [inline server/controllers/content/content.ts:17]: export enum DOWNLOAD_STATUS {
    SUBMITTED = "DOWNLOADING",
    COMPLETED = "DOWNLOADING",
    EXTRACTED = "DOWNLOADING",
    INDEXED = "DOWNLOADED",
    FAILED = "FAILED"
}

File: server/controllers/content/content.ts [hunk 4]
```diff
@@ -416,15 +429,15 @@ export default class Content {
...
                 .then(data => {
-                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in ContentDb`)
+                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in Content_Download Db`)
                     for (let doc of data.docs) {
                         for (let content of contents) {
-                            logger.debug(`include addedToLibrary property for the contents which are downloaded`)
-                            this.includeAddedToLibraryProperty(doc, content, reqId);
-                            logger.info(`ReqId = "${reqId}": included addedToLibrary property for the contents which are downloaded`)
+                            logger.debug(`add downloadStatus for the contents`)
+                            this.includeDownloadStatus(doc, content, reqId);
```
Comment (anoophm) [inline server/controllers/content/content.ts]: if(doc.contentId === content.identifier && doc.status !== CONTENT_DOWNLOAD_STATUS.Failed){
content.downloadStatus
}
// TODO: will handle failed case
add above comment also

File: server/controllers/content/content.ts [hunk 5]
```diff
@@ -464,25 +477,32 @@ export default class Content {
...
-                logger.info(`ReqId = "${reqId}": updated Content in ContentDb`)
-            } catch (err) {
-                console.log(err);
-                logger.error(
-                    `ReqId = "${reqId}": Received error while updating content in database and err.message: ${
-                    err.message
-                    } ${err}`
-                );
-            }
+    includeDownloadStatus(doc, content, reqId) {
```
Comment (anoophm) [inline server/controllers/content/content.ts]: not needed

File: server/controllers/content/content.ts [hunk 6]
```diff
@@ -464,25 +477,32 @@ export default class Content {
...
+    /* This method is to search contents for download status in database  */
+
+    searchDownloadingContent(contents, reqId) {
+        logger.debug(`ReqId = "${reqId}": searchDownloadingContent method is called`);
+        let dbFilters =  {
+          "selector": {
+              "contentId": {
+                  "$in": contents
+              },
+              "createdOn": {
```
Comment (anoophm) [inline server/controllers/content/content.ts]: do we need createdOn

File: server/controllers/content/content.ts [hunk 7]
```diff
@@ -14,9 +14,19 @@ import Hashids from "hashids";
...
+    Submitted = "SUBMITTED",
+    Completed = "COMPLETED",
+    Extracted = "EXTRACTED",
+    Indexed = "INDEXED",
+    Failed = "FAILED"
+}
 export default class Content {
     private contentsFilesPath: string = 'content';
     private ecarsFolderPath: string = 'ecars';
+    private downloaded;
```
Comment (anoophm) [inline server/controllers/content/content.ts]: not needed

File: server/controllers/content/content.ts [hunk 8]
```diff
@@ -14,9 +14,19 @@ import Hashids from "hashids";
...
+    private downloading;
```
Comment (anoophm) [inline server/controllers/content/content.ts]: not needed

File: server/controllers/content/content.ts [hunk 9]
```diff
@@ -14,9 +14,19 @@ import Hashids from "hashids";
...
+    private failed;
```
Comment (anoophm) [inline server/controllers/content/content.ts]: not needed

File: server/controllers/content/content.ts (hunk 3 above)
Comment (HarishGangula) [reply server/controllers/content/content.ts:17]: import this enum from content_download file

File: server/controllers/content/content.ts [hunk 10]
```diff
@@ -415,16 +422,15 @@ export default class Content {
...
                 .then(data => {
-                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in ContentDb`)
+                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in Content_Download Db`)
                     for (let doc of data.docs) {
                         for (let content of contents) {
-                            logger.debug(`include addedToLibrary property for the contents which are downloaded`)
-                            this.includeAddedToLibraryProperty(doc, content, reqId);
-                            logger.info(`ReqId = "${reqId}": included addedToLibrary property for the contents which are downloaded`)
+                            if (doc.contentId === content.identifier) {
+                                content.downloadStatus = DOWNLOAD_STATUS[doc.status];
```
Comment (HarishGangula) [inline server/controllers/content/content.ts:432]: don't send or have a hard dependency from backend status with the front end, have some abstraction.

**Review by HarishGangula — CHANGES_REQUESTED** [review]

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**